goss/runner.go

```diff
@@ -0,0 +1,103 @@
+package goss
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/radiofrance/dib/types"
+)
+
+var errGossCommandFailed = errors.New("goss command failed")
+
+// Executor is an interface for executing goss tests.
+type Executor interface {
+	Execute(ctx context.Context, output io.Writer, opts types.RunTestOptions, args ...string) error
+}
+
+// TestRunner implements types.TestRunner.
+type TestRunner struct {
+	Executor
+	TestRunnerOptions
+}
+
+// TestRunnerOptions are the configuration options for TestRunner.
+type TestRunnerOptions struct {
+	ReportsDirectory string
+	WorkingDirectory string
+	JUnitReports     bool
+}
+
+// NewTestRunner creates a new instance of TestRunner.
+func NewTestRunner(executor Executor, opts TestRunnerOptions) *TestRunner {
+	return &TestRunner{executor, opts}
+}
+
+// Supports returns true if a goss.yml file is found at the target context path.
+func (b TestRunner) Supports(opts types.RunTestOptions) bool {
+	if _, err := os.Stat(path.Join(opts.DockerContextPath, "goss.yaml")); err != nil {
```

Comment:
Can we handle both `goss.yml` and `goss.yaml` ? You did the mistake, you have different extensions in the code and in your comment

Reply:
ok for me, just need to update your comment :)